includes/data-events/connectors/ga4/class-ga4.php

```diff
@@ -88,6 +100,31 @@ public static function handle_reader_logged_in( $timestamp, $data, $user_id ) {
 		self::send_event( $event, $client_id, $timestamp, $user_id );
 	}
 
+	/**
+	 * Handler for the reader_registered event.
+	 *
+	 * @param int   $timestamp Timestamp of the event.
+	 * @param array $data      Data associated with the event.
+	 * @param int   $user_id   ID of the client that triggered the event. It's a RAS ID.
+	 *
+	 * @throws \Exception If the event is invalid.
+	 * @return void
+	 */
+	public static function handle_reader_registered( $timestamp, $data, $user_id ) {
+
+		$params    = $data['ga_params'];
+		$client_id = $data['ga_client_id'];
+
+		if ( empty( $client_id ) ) {
+			throw new \Exception( 'Missing client ID' );
+		}
+
+		$params['registration_method'] = $data['metadata']['registration_method'] ?? '';
+
+		$event = new Event( 'reader_registered', $params );
+		self::send_event( $event, $client_id, $timestamp, $user_id );
```

Comment:
In the context of a global handler for analytics, it can look like this:

_Being a global hanler, this suggestion assumes the first argument of the method is the `$action_name`._

```suggestion
		$event = new Event( $action_name, $data );
		self::send_event( $event, $client_id, $timestamp, $user_id );
```

Reply:
I cant' simply pass whatever is in `$data` to GA4. Each event will have its own handler to add the appropriate parameters

Reply:
I wish we could. The listeners' documentation would be the source of truth on what and how we collect and we avoid having another layer of support for each new trackable thing. The more we track the better, there's no downside to that IMO.

Reply:
There's a limit of 25 parameters an event can have in GA, so I think we want to carefully choose what we will track.

Also, some events have arrays as parameters, that need to be treated...